src/subset.rs

```diff
@@ -295,15 +314,19 @@ impl<N: NodeIdT + Rand> Subset<N> {
         self.ba_results.values().filter(|v| **v).count()
     }
 
-    fn try_binary_agreement_completion(&mut self) -> Option<BTreeMap<N, ProposedValue>> {
+    fn try_binary_agreement_completion(
+        &mut self,
+        proposer_id: &N,
+        result: Option<Vec<u8>>,
+    ) -> Option<SubsetOutput<N>> {
```

Comment:
Maybe the "true" count check in this method should be moved further down, at least if we now want to output `Contribution`s (but not `Done`) even before we have `num_correct()` instances that returned `true`.

In any case, we need to make sure that for each accepted value (where the agreement instance returns `true`), we output `Contribution` exactly once. Note that this can be triggered by two things:
* An agreement instance outputs `true` and we already have the value—the output of the broadcast instance—for it.
* A broadcast instance outputs a value and we already have received the output `true` from the corresponding agreement instance. (Not very likely, but it can happen if our node is lagging.)